main_test.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"testing"
+)
+
+func TestStartServer(t *testing.T) {
+	other := "https://example.com"
+	letsEncrypt := fmt.Sprintf("https://%s", letsEncryptStaging)
+
+	for _, tt := range []struct {
+		allowAll bool
+		dst      string
+		success  bool
+	}{
+		{allowAll: false, dst: letsEncrypt, success: true},
+		{allowAll: false, dst: other, success: false},
+		{allowAll: true, dst: letsEncrypt, success: true},
+		{allowAll: true, dst: other, success: true},
+	} {
+		cfg := config{
```

Comment:
one thing to note that these tests do make http calls. Since this tool is so simple, really isolating the levels to support unit testing seemed like overkill.  Alt, I can change to so that we just test the AllowCalls for the rule sets (but personally,  I think having some integration tests is not a terrible thing).